packages/ui/src/ui/pages/navigation/tabs/Flow/FlowLayout/FlowLayout.tsx

```diff
@@ -0,0 +1,208 @@
+import React from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import cn from 'bem-cn-lite';
+
+import {Column} from '@gravity-ui/react-data-table';
+
+import format from '../../../../../common/hammer/format';
+
+import ClickableAttributesButton from '../../../../../components/AttributesButton/ClickableAttributesButton';
+import ClipboardButton from '../../../../../components/ClipboardButton/ClipboardButton';
+import DataTableYT, {
+    DATA_TABLE_YT_SETTINGS,
+} from '../../../../../components/DataTableYT/DataTableYT';
+import ErrorBlock from '../../../../../components/Error/Error';
+import {ExpandButton} from '../../../../../components/ExpandButton';
+
+import {useUpdater} from '../../../../../hooks/use-updater';
+import {
+    expandFlowLayoutComputation,
+    expandFlowLayoutWorker,
+    loadFlowLayout,
+} from '../../../../../store/actions/flow/layout';
+import {
+    FlowLayoutDataItem,
+    getFlowLayoutData,
+    getFlowLayoutError,
+    getFlowLayoutPipelinePath,
+} from '../../../../../store/selectors/flow/layout';
+
+import './FlowLayout.scss';
+
+const block = cn('yt-navigation-flow-layout');
+
+export function FlowLayout({path, viewMode}: {path: string; viewMode: 'computations' | 'workers'}) {
+    const dispatch = useDispatch();
+
+    const pipeline_path = useSelector(getFlowLayoutPipelinePath);
+    const error = useSelector(getFlowLayoutError);
+    const getData = useSelector(getFlowLayoutData);
+
+    const data = React.useMemo(() => {
+        return getData(viewMode);
+    }, [viewMode, getData]);
+
+    const samePath = path === pipeline_path;
+    const columns = useFlowLayoutColumn(viewMode);
+
+    const updateFn = React.useCallback(() => {
+        dispatch(loadFlowLayout(path));
+    }, [path, dispatch]);
+    useUpdater(updateFn);
+
+    if (!samePath) {
+        return null;
+    }
+
+    return (
+        <div className={block()}>
+            {Boolean(error) && <ErrorBlock error={error} />}
+            <DataTableYT
+                settings={DATA_TABLE_YT_SETTINGS}
+                columns={columns}
+                data={data}
+                useThemeYT
+            />
+        </div>
+    );
+}
+
+function useFlowLayoutColumn(type: 'computations' | 'workers') {
+    const dispatch = useDispatch();
+
+    const res = React.useMemo(() => {
+        const isComputations = type === 'computations';
+        const columns: Array<Column<FlowLayoutDataItem>> = [
+            {
+                name: isComputations
+                    ? 'Computation / Partition Id'
+                    : 'Worker address / Partition Id',
+                className: block('td-name'),
+                render({row}) {
+                    let content;
+                    if ('$attributes' in row) {
+                        const {
+                            $attributes: {name, expanded},
+                        } = row;
+                        content = (
+                            <>
```

Comment:
let's put it in a separate component

Reply:
Such component will not be reusable. I hope you don't mind to leave in current state.